tracking-with-geofence-notifications/app/src/main/java/software/amazon/location/sample/viewModel/MainViewModel.kt

```diff
@@ -60,9 +52,8 @@ class MainViewModel : ViewModel() {
     var lastAccuracyMeasured by mutableStateOf("")
     var isConfigDialogVisible: Boolean by mutableStateOf(false)
     var selectedTrackingMode = BackgroundTrackingMode.ACTIVE_TRACKING
-    var amazonLocationClient: AmazonLocationClient? = null
-    var mCognitoCredentialsProvider: CognitoCredentialsProvider? = null
     var enableGeofences = false
+    var locationCredentialsProvider: LocationCredentialsProvider? = null
```

Comment:
Is there a way to handle this case without setting any values as null?

Reply:
Code optimized. For null, the above approach is widely used, and also I am using the safe (?.) operator while accessing to avoid crashes.